src/librustc_driver/args/mod.rs

```diff
@@ -0,0 +1,105 @@
+use std::env;
+use std::error;
+use std::fmt;
+use std::fs;
+use std::io::{self, BufRead};
+use std::str;
+use std::sync::atomic::{AtomicBool, Ordering};
+
+#[cfg(test)]
+mod tests;
+
+static USED_ARGSFILE_FEATURE: AtomicBool = AtomicBool::new(false);
+
+pub fn used_unstable_argsfile() -> bool {
+    USED_ARGSFILE_FEATURE.load(Ordering::Relaxed)
+}
+
+struct FileArgs {
+    path: String,
+    input: Vec<u8>,
+}
+
+impl FileArgs {
+    fn new(path: String, input: Vec<u8>) -> Self {
+        FileArgs { path, input }
+    }
+
+    fn lines(self) -> impl Iterator<Item = Result<String, Error>> {
+        let Self { input, path } = self;
+        io::Cursor::new(input).lines().map(move |res| {
```

Comment:
I don't think `Cursor` should be necessary here, `&[u8]` already implements `BufRead` so `BufRead::lines(input)` should suffice here

Reply:
I couldn't make this work because it ends up needing a self-borrow to keep the nested iterator structure.